### PR TITLE
add pytest plumbing

### PR DESCRIPTION
**Pull request by jhoblitt**
This adds easy execution of {{pytest/flake8}} during development and lays the foundation for unit testing.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+with open(path.join(here, 'README'), encoding='utf-8') as f:
+    long_description = f.read()
+
+setup(
+    name='lsst_build',
+    version='1.0.0',
+    description='LSST build automation',
+    long_description=long_description,
+    url='https://github.com/lsst/lsst_build',
+    author='LSST DM',
```
Comment (timj) [inline setup.py]: Maybe say "Large Synoptic Survey Telescope Data Management Team" here.

File: .travis.yml [hunk 2]
```diff
@@ -3,11 +3,25 @@ language: python
...
-      python: 3.5
+      python: '3.6'
       env: DEPLOY_ARGS="-3 -b"
     - os: linux
-      python: 2.7
+      python: '2.7'
       env: DEPLOY_ARGS="-2 -b"
+    - env: TEST='pytest'
+      python: '3.6'
+      install: pip install --upgrade 'pip>=9' 'pytest>=3,<4' 'flake8>=3,<4'
```
Comment (timj) [inline .travis.yml]: Can we use a `requirements.txt` file so that we don't duplicate these versions and dependencies in three places?

**Review by timj — APPROVED** [review]
Thanks for this. It's good to check this.

File: .travis.yml (hunk 2 above)
Comment (jhoblitt) [reply .travis.yml]: The reason I didn't is that `flake8` isn't required to run `pytest|python setup.py test`.  For a long time, github was shipping an old `pip` version in their `2.7` env that was choking on version constraints between certain pypi packages.  I also vaguely recall there was some sort dep resolution problem with an old version of `pytest` being preinstalled -- so I cut'n'pasted from a sqre pypi package.

Comment (jhoblitt) [reply .travis.yml]: I've added a `.travis.requirements.txt` and am testing if `.travis.yaml` can be DRY'd up with yaml anchors.

Comment (jhoblitt) [reply .travis.yml]: I've managed to condense it down to:
```
    - &pytest
      env: TEST='pytest'
      python: '3.6'
      install: pip install --upgrade -r .travis.requirements.txt
      script: |
        python setup.py test
        # https://github.com/tholo/pytest-flake8/issues/27
        flake8 bin/lsst-build
    - <<: *pytest
      python: '2.7'
```

**Pull request merged**